File: inventory_management/src/main/java/com/inventory/inventory_management/entities/Order.java
package com.inventory.inventory_management.entities;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

@Entity
@Table(name = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long orderId;

    private String orderCode;

    private BigDecimal totalAmount;
    private BigDecimal discount;
    private BigDecimal actualAmount;

    private LocalDateTime createdAt;

    private String status;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonBackReference("user-orders")
    private User user;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("order-orderItems")
    private List<OrderItem> orderItems;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("order-transactions")
    private List<Transaction> transactions;
}

File: mobile/inventory_sales_app/lib/models/order.dart
import 'package:inventory_sales_app/models/order_items.dart';

class Order {
  final String userId;
  final double totalAmount;
  final double discount;
  final double actualAmount;
  final List<OrderItem> orderItems;

  Order({
    required this.userId,
    required this.totalAmount,
    this.discount = 0.0,
    required this.actualAmount,
    required this.orderItems,
  });

  Map<String, dynamic> toJson() => {
        "userId": userId,
        "totalAmount": totalAmount,
        "discount": discount,
        "actualAmount": actualAmount,
        "orderItems": orderItems.map((item) => item.toJson()).toList(),
      };
}
